src/Cli/dotnet/Commands/Package/PackageCommandParser.cs

```diff
@@ -34,4 +46,20 @@ public static Command GetCommand()
 
         return command;
     }
-} 
+
+    public static (string Path, AppKinds AllowedAppKinds) ProcessPathOptions(ParseResult parseResult)
+    {
+        bool hasFileOption = parseResult.HasOption(FileOption);
+        bool hasProjectOption = parseResult.HasOption(ProjectOption);
+
+        return (hasFileOption, hasProjectOption) switch
+        {
+            (false, false) => parseResult.GetValue(ProjectOrFileArgument) is { } projectOrFile
+                ? (projectOrFile, AppKinds.Any)
+                : (Environment.CurrentDirectory, AppKinds.ProjectBased),
+            (true, false) => (parseResult.GetValue(FileOption)!, AppKinds.FileBased),
+            (false, true) => (parseResult.GetValue(ProjectOption)!, AppKinds.ProjectBased),
```

Comment:
So is the "optionality" of an `Option<T>` indicated by giving it a nullable type argument? I am just curious if something would change for the worse here if `FileOption` and `ProjectOption` were `Option<string>`.